### ISSUE TITLE
Failed to format code: No code formatting providers found!

### ISSUE THREAD

**Issue by sweeneychris** [facebook/nuclide#1248]
### Issue and Steps to Reproduce
Open a c++ file and run Edit -> Text -> Format Code

### Expected Behavior
Code should be formatted with clang-format

### Actual Behavior
An error occurs stating: "Failed to format code: No code formatting providers found!"

### Versions
* Atom: 1.18.0
* Nuclide: 0.242.0
* Client OS: Mac OS X
* Server OS (optional): 

### Additional Details
* Installed packages (`apm ls --installed`):
Community Packages (40) /Users/csweeney/.atom/packages
├── MagicPython@1.0.11
├── atom-beautify@0.30.4
├── atom-material-syntax-dark@1.0.0
├── atomic-emacs@0.11.0
├── autocomplete-clang@0.11.3
├── brahalla-syntax@0.5.0
├── build@0.68.0
├── build-cmake@0.10.1
├── busy@0.7.0
├── busy-signal@1.4.3
├── clang-format@1.29.0
├── emacs-tab@0.1.0
├── file-icons@2.1.9
├── haskell-grammar@0.4.0
├── highlight-selected@0.13.1
├── intentions@1.1.5
├── language-babel@2.68.0
├── language-cmake@0.3.0
├── language-cpp14@0.6.2
├── language-cuda@0.1.0
├── language-google-cstyle@2.0.0
├── language-graphql@0.9.0
├── language-ini@1.19.0
├── language-lua@0.9.11
├── language-ocaml@1.9.5
├── language-rust@0.4.11
├── language-swift@0.5.0
├── language-thrift@1.0.2
├── linter@2.2.0
├── linter-cpplint@2.0.1
├── linter-ui-default@1.6.3
├── nebula-syntax@0.4.4
├── nebula-ui@0.6.0
├── nuclide@0.242.0
├── nuclide-format-js@0.0.36
├── one-dark-vivid-syntax@1.8.0
├── set-syntax@0.3.2
├── sort-lines@0.14.0
├── sync-settings@0.8.2
└── tool-bar@1.1.0

**Comment by shushz** [comment]
Nuclide actually does run `clang-format` for C++ grammar files provided it's on its path.
What was the active grammar on your file? The one shown in the right part of the bottom status bar.

**Comment by sweeneychris** [comment]
Yes, I know it should run clang-format as noted in the "expected behavior" in my issue. The problem is that it instead shows an error stating: "Failed to format code: No code formatting providers found!"

Yes, I can verify it is detecting c++ files with the correct grammar (status bar shows c++ grammar)

**Comment by lrn2prgrm** [comment]
I am suffering the same issue.

**Comment by henry-p** [comment]
Same, trying to format Babel/JSX ES6 code.

**Comment by tymarats** [comment]
Same, trying to format Babel/JSX ES6 code.

**Comment by ThisIsRuddy** [comment]
Anyone ever figure this out?

**Comment by lenhhoxung32** [comment]
Got the same issue

**Comment by mosaic141688** [comment]
I had the same issue when trying to format vuejs

**Comment by ir2pid** [comment]
same same

**Comment by pbbhopp** [comment]
Got same issue, even atom-beauty is installed.

**Comment by ayttunc** [comment]
same here, no solutions yet ?
